src/nvim/ops.c

```diff
@@ -2497,9 +2499,7 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
    */
   curbuf->b_op_start = oap->start;
   curbuf->b_op_end = oap->end;
-  if (yanktype == MLINE
-      && !oap->block_mode
-      ) {
+  if (yanktype == MLINE) {
```

Comment:
You're right that oap->block_mode could have be true even if oap->motion_type == MLINE, if a linewise motion get forced into block-mode (`d^V3d`). But then the operation is still purely block-wise, the fact that it "once was" a linewise motion has no effect. (I mean, this if block will be executed under the same set of circumstances as before, even if the condition _looks_ different)